### ISSUE TITLE
[BUG] Unsure if motion Vectors are working as i expected

### ISSUE THREAD

**Issue by PlasmaDev5** [SparkyStudios/ezEngine#7]
This is the output of motion vectors under movement.
![image](https://user-images.githubusercontent.com/3044415/210289176-1f625a8b-97bf-4327-9515-c2dc98ad531a.png)

Velocity source texture is set to RGBAHalf 
Using Deferred -> Sky ->  Transparent -> Motion Blur (After Sharpening)

**Comment by na2axl** [comment]
The Velocity source doesn't need to be rgbaHalf, just rgHalf is enough.

Have you tried to update the velocity strength in the pass parameters? Be aware the final motion vector strength will be scaled with the camera's ShutterTime and the frame's delta time too. Also, what do you mean by "as I expected"? I'm curious to know what you wanted as output.

**Comment by PlasmaDev5** [comment]
In my branch im currently packing some of the clearcoat data into the alpha so i dont have to make an additional buffer. 
From testing it seams to be harmless enough unless its somehow whats effecting motion vectors but i doubt it.
The main thing of concern is the screen space position gradient (red left, green top)
i was kinda more expecting something like the image below but without the arrows, color representing direction, brightness for strength
![image](https://user-images.githubusercontent.com/3044415/210346085-d7931bf0-8cf8-4d00-ac76-dd466c8dfb06.png)

**Comment by na2axl** [comment]
Ok got it...

First, I'm not sure but these values seem to be a bit exaggerated for me, maybe the renderer used to create your image has scaled the vectors to make them noticeable on screen? Because AFAIK for the motion blur effect, motion vectors should be really small to not distort the final image.

Secondly the motion vectors in the image seem to be in 3D space (I'm saying that because of the purple-blue variation). But the motion blur is building his motion vectors on the screen space, you can only have black, red, green and yellow colors. And also the color space variation is not static and depends on the camera position.

![image](https://user-images.githubusercontent.com/14062034/210369132-a1cdc013-36cb-4b94-bc34-884d1626f4ec.png)

![image](https://user-images.githubusercontent.com/14062034/210369806-bab71240-c129-4e3f-a275-215e2cbeb618.png)

I feel like what you want instead is vertex velocity?

**Comment by PlasmaDev5** [comment]
is it normal that i still have the screenspace gradient even when everything is static
![image](https://user-images.githubusercontent.com/3044415/210461712-a0938193-0a91-400b-b089-bc7e41273ebf.png)

**Comment by na2axl** [comment]
No surely not, can you please share your render pipeline?

**Comment by PlasmaDev5** [comment]
![image](https://user-images.githubusercontent.com/3044415/210462055-045fd1db-4c54-4c23-9e8d-605277dd6674.png)

**Comment by na2axl** [comment]
Just reproduced your entire pipeline, but I'm not having this issue, even deactivated SourcePass clear but everything is working. Is this actually affecting the motion blur effect? Also did the velocity texture in renderdoc start with the output of the old frame?

Also, if you are having this issue in the editor, I've been through it a bit, It seems to be due to the fact the viewport got initialized with the size defined in Window settings and then resized to the Editor's viewport size. A quick trick is just to resize the viewport again, even a little, and the velocity map shows correct values.

But for in-game windows it should not happen.

**Comment by PlasmaDev5** [comment]
Looks like it was the issue with the editor resizing :D it was effecting the motion blur but now i have resized it looks to be working fine

**Comment by PlasmaDev5** [comment]
Would this also effect the upscale pass because it errors something about texture size == 0

**Comment by na2axl** [comment]
Yes it can be the cause, but only for the very first frame I think. But the upscale pass is not meant to be used by the editor. And I've done nothing to help in that 😅 

This is a game-only feature, since even the scaling ratio (calculated from the selected preset) will be precomputed and hence not usable on sizeable windows...

**Comment by PlasmaDev5** [comment]
Would be very useful for the editor as it would allow developers on weaker machines to get that little extra performance. I feel it makes sense to have the editor viewport match the game as close as possible
Ill leave this open until the editor stuff for motion blur is resolved

**Comment by PlasmaDev5** [comment]
Also looks to also effect in game with the same issue for motion blur although maybe a little stronger
